bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/Rules.java

```diff
@@ -43,6 +43,8 @@ class Rules implements IResourceRuleFactory, ILifecycleListener {
 	 * @param workspace
 	 */
 	Rules(Workspace workspace) {
+		this.defaultFactory = new InternalResourceRuleFactory(workspace) {
+			/**/};
```

Comment:
Please remove ` {/**/}`. Why do we need an inner class here?

Reply:
This is required because ResourceRulefactory has only protected constructors....